core/lib/types/src/transaction_request.rs

```diff
@@ -818,9 +818,7 @@ impl L2Tx {
         validate_factory_deps(&meta.factory_deps)?;
 
         let mut tx = L2Tx::new(
-            value
-                .to
-                .ok_or(SerializationTransactionError::ToAddressIsNull)?,
+            value.to,
```

Comment:
so this means we start allowing txs with to=null in our system but process them as if `to=0x00..00`. I would continue rejecting such txs until evm equivalence is on, otherwise it will be confusing for everyone

Reply:
Done! I added the check above and included a comment for its removal once the EVM interpreter is integrated.